core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -528,7 +522,7 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
     override def requestAndWaitForCommit() {
       this.synchronized {
         // only wait() if mirrorMakerConsumer has been initialized and it has not been cleaned up.
-        if (iter != null) {
+        while (iter != null) {
```

Comment:
This is to protect against spurious wake ups, right? `iter` is not null after initialisation, so it doesn't seem like it works for that purpose. Seems like `immediateCommitRequested` could be that if we do something like:

```scala
if (iter != null) {
  immediateCommitRequested = true
  while (!immediateCommitRequested)
    this.wait()
```

But please check if this makes sense.